### DISCUSSION TITLE
Add ability to only apply compound variant if variant value is not the specified value

### DISCUSSION THREAD

**Discussion by AsyncBanana** [joe-bell/cva#244] [Ideas]
Currently, as far as I can tell, there is no way to simply invert a condition in a compound variant. The only ways to get around this are by overriding other classes or listing every other variant. It would be helpful if there was an API like this instead:

```js
import { cva, not } from "cva";

const button = cva({
  base: "font-semibold border rounded",
  variants: {
    intent: {
      primary: "bg-blue-500 text-white border-transparent hover:bg-blue-600",
      secondary: "bg-white text-gray-800 border-gray-400 hover:bg-gray-100",
    },
    size: {
      small: "text-sm py-1 px-2",
      medium: "text-base py-2 px-4",
    },
  },
  compoundVariants: [
    {
      intent: not("primary"), // equivalent to ["secondary"] in this case
      size: "medium",
      class: "uppercase",
    },
  ],
  defaultVariants: {
    intent: "primary",
    size: "medium",
  },
});
```

Is there another way to currently do this that I am not aware of? I would be happy to work on a PR to add this if necessary.

**Comment by joe-bell** [comment]
Unfortunately not yet, but it's on my list of potential improvements!

**Comment by AsyncBanana** [comment]
Would you be open to a pull request adding this?
